### ISSUE TITLE
PluginImageHelper leaks images

### ISSUE THREAD

**Issue by iloveeclipse** [eclipse/xtext-eclipse#1621]
Regression from #432 : if PluginImageHelper can find a descriptor, it creates images without adding them to the registry & so without re-using them, causing permanent image leak:

```
org.eclipse.xtext.ui.PluginImageHelper.getImage(String)
	public Image getImage(String imageName) {
		final ImageDescriptor descriptor = findImage(imageName);
		if (descriptor != null) {
			return descriptor.createImage(); // The image is leaked here!
		}
```

**Comment by iloveeclipse** [comment]
@cdietrich : FYI

**Comment by cdietrich** [comment]
the parameter is

`-Dorg.eclipse.swt.graphics.Resource.reportNonDisposed=true`

**Comment by cdietrich** [comment]
i also saw, but cannot consistently reproduce

```
java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:824)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:338)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:287)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:265)
	at org.eclipse.xtext.ui.PluginImageHelper.getImage(PluginImageHelper.java:87)
	at org.eclipse.xtext.ui.editor.XtextEditorErrorTickUpdater$1.run(XtextEditorErrorTickUpdater.java:148)
	at org.eclipse.swt.widgets.RunnableLock.run(RunnableLock.java:40)
	a
```

```
java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:771)
	at org.eclipse.jface.resource.URLImageDescriptor.createImage(URLImageDescriptor.java:274)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:287)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:265)
	at org.eclipse.xtext.ui.PluginImageHelper.getImage(PluginImageHelper.java:87)
	at org.eclipse.xtext.ui.label.AbstractLabelProvider.convertToImage(AbstractLabelProvider.java:96)
	at org.eclipse.xtext.ui.label.AbstractLabelProvider.getImage(AbstractLabelProvider.java:59)
	at org.eclipse.jface.viewers.WrappedViewerLabelProvider.getImage(WrappedViewerLabelProvider.java:104)
	at org.eclipse.jface.viewers.WrappedViewerLabelProvider.update(WrappedViewerLabelProvider.java:149)
	at org.eclipse.jface.viewers.ViewerColumn.refresh(ViewerColumn.java:144)
	at org.eclipse.jface.viewers.AbstractTreeViewer.doUpdateItem(AbstractTreeViewer.java:959)
	at org.eclipse.jface.viewers.AbstractTreeViewer$UpdateItemSafeRunnable.run(AbstractTreeViewer.java:126)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.AbstractTreeViewer.doUpdateItem(AbstractTreeViewer.java:1037)
	at org.eclipse.jface.viewers.StructuredViewer$UpdateItemSafeRunnable.run(StructuredViewer.java:427)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:45)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:174)
	at org.eclipse.jface.viewers.StructuredViewer.updateItem(StructuredViewer.java:2111)
	at org.eclipse.jface.viewers.AbstractTreeViewer.createTreeItem(AbstractTreeViewer.java:853)
	at org.eclipse.jface.viewers.AbstractTreeViewer.createChildren(AbstractTreeViewer.java:831)
	at org.eclipse.jface.viewers.TreeViewer.createChildren(TreeViewer.java:604)
	at org.eclipse.jface.viewers.AbstractTreeViewer.createChildren(AbstractTreeViewer.java:779)
	at org.eclipse.jface.viewers.AbstractTreeViewer.internalInitializeTree(AbstractTreeViewer.java:1599)
	at org.eclipse.jface.viewers.TreeViewer.internalInitializeTree(TreeViewer.java:780)
```

and this one can be consistently reproduced.
wonder why we use the new here cc @miklossy 

```
java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Image.<init>(Image.java:771)
	at org.eclipse.jface.resource.URLImageDescriptor.createImage(URLImageDescriptor.java:274)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:287)
	at org.eclipse.jface.resource.ImageDescriptor.createImage(ImageDescriptor.java:265)
	at org.eclipse.xtext.ui.PluginImageHelper.getImage(PluginImageHelper.java:114)
	at org.eclipse.xtext.example.fowlerdsl.ui.wizard.MrsGrantsSecretCompartmentsProject.image(MrsGrantsSecretCompartmentsProject.java:103)
	at org.eclipse.xtext.example.fowlerdsl.ui.wizard.MrsGrantsSecretCompartmentsProject.getImages(MrsGrantsSecretCompartmentsProject.java:92)
	at org.eclipse.xtext.ui.wizard.template.NewProjectWizardTemplateSelectionPage.createControl(NewProjectWizardTemplateSelectionPage.java:93)
```

**Comment by homebeaver** [comment]
Hi,
here another trace:
```
java.lang.Error: SWT Resource was not properly disposed
	at org.eclipse.swt.graphics.Resource.initNonDisposeTracking(Resource.java:172)
	at org.eclipse.swt.graphics.Resource.<init>(Resource.java:120)
	at org.eclipse.swt.graphics.Font.<init>(Font.java:111)
	at org.eclipse.buildship.ui.internal.util.font.FontUtils.getDefaultDialogFont(FontUtils.java:34)
	at org.eclipse.buildship.ui.internal.util.widget.GradleDistributionGroup.createWidgets(GradleDistributionGroup.java:95)
	at org.eclipse.buildship.ui.internal.util.widget.GradleDistributionGroup.<init>(GradleDistributionGroup.java:86)
	at org.eclipse.buildship.ui.internal.util.widget.GradleProjectSettingsComposite.createGradleDistributionGroup(GradleProjectSettingsComposite.java:100)
	at org.eclipse.buildship.ui.internal.util.widget.GradleProjectSettingsComposite.<init>(GradleProjectSettingsComposite.java:66)
	at org.eclipse.buildship.ui.internal.util.widget.GradleProjectSettingsComposite.<init>(GradleProjectSettingsComposite.java:38)
	at org.eclipse.buildship.ui.internal.util.widget.GradleProjectSettingsComposite$GradleProjectSettingsCompositeBuilder.build(GradleProjectSettingsComposite.java:253)
	at org.eclipse.buildship.ui.internal.preferences.GradleWorkbenchPreferencePage.createContents(GradleWorkbenchPreferencePage.java:61)
	at org.eclipse.jface.preference.PreferencePage.createControl(PreferencePage.java:244)
	at org.eclipse.jface.preference.PreferenceDialog.createPageControl(PreferenceDialog.java:1433)
	at org.eclipse.jface.preference.PreferenceDialog$8.run(PreferenceDialog.java:1196)
...
```
Session data:
eclipse.buildId=4.20.0.I20210611-1600
java.version=11
java.vendor=Oracle Corporation
BootLoader constants: OS=win32, ARCH=x86_64, WS=win32, NL=de_DE
Framework arguments:  -product org.eclipse.epp.package.java.product -product org.eclipse.epp.package.java.product
Command-line arguments:  -os win32 -ws win32 -arch x86_64 -product org.eclipse.epp.package.java.product -data file:/C:/proj/eclipseWS/16-edocs/ -product org.eclipse.epp.package.java.product

![grafik](https://user-images.githubusercontent.com/3493126/122612686-d7845800-d083-11eb-972d-23ca45fdcc52.png)

**Comment by cdietrich** [comment]
@homebeaver buildship problems should be reported here https://github.com/eclipse/buildship/issues

**Comment by cdietrich** [comment]
@miklossy  can you answer the question about the `new PluginImageHelper`

**Comment by cdietrich** [comment]
created https://github.com/eclipse/xtext-eclipse/pull/1690 for that

**Comment by cdietrich** [comment]
Fixed in 2.26
